build_all/base-libraries/AzureIoTUtility/src/esp32/azcpgmspace.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/** @file azcpgmspace.h
+*	@brief Function prototypes for pgmspace functions that need extern-ing to C
+*
+*	@details These fucntions are just wrappers around existing ones in pgmspace.h that
+*    are not defined in a way to make them linkable from c libs.
+*/
+#ifndef ARDUINO_ARCH_ESP8266
+#ifndef AZCPGMSPACE_H
+#define AZCPGMSPACE_H
+
+#include <pgmspace.h>
+#include "azure_c_shared_utility/crt_abstractions.h"
+
+
+#ifdef __cplusplus
+#include <cstddef>
+extern "C" {
+#else
+#include <stddef.h>
+#endif
+extern char* az_c_strncpy_P(char* dest, PGM_P src, size_t size);
+extern size_t az_c_strlen_P(PGM_P s);
+#ifdef __cplusplus
+}
+#endif
+
+#endif // _AZCPGMSPACE_H
+
+#endif //
```

Comment:
comment missing // ARDUINO_ARCH_ESP8266

Still wondering why you want to enable this header and implementation for all other then ARDUINO_ARCH_ESP8266. Is this what you intend?

Reply:
Nope, changed to ARDUINO_ARCH_ESP32.

Reply:
Can you please close things which you have take care of?

---
In reply to: [350979001](https://github.com/Azure/azure-iot-pal-arduino/pull/7#discussion_r350979001) [](ancestors = 350979001)